client/me/account/main.jsx

```diff
@@ -72,22 +74,23 @@ const colorSchemeKey = 'calypso_preferences.colorScheme';
  */
 const debug = debugFactory( 'calypso:me:account' );
 
+const ALLOWED_USERNAME_CHARACTERS_REGEX = /^[a-z0-9]+$/;
+
 /* eslint-disable react/prefer-es6-class */
 const Account = createReactClass( {
 	displayName: 'Account',
 
 	// form-base mixin is needed for getDisabledState() (and possibly other uses?)
-	mixins: [ formBase, observe( 'userSettings', 'username' ) ],
+	mixins: [ formBase, observe( 'userSettings' ) ],
```

Comment:
It'd be awesome if we could move away from using `createReactClass` and avoid those mixins. At least `formBase` is used in a bunch of places and would probably require a more thoughtful refactor.